src/test/java/com/commercetools/emailprocessor/email/EmailProcessorTest.java

```diff
@@ -0,0 +1,223 @@
+import org.apache.commons.io.IOUtils;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.*;
+import java.net.*;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+import static org.junit.Assert.*;
+
+
+public class EmailProcessorTest {
+
+
+    public EmailProcessor emailProcessor = null;
+    TenantConfiguration tenantConfiguration = null;
+    List<CustomObject<Email>> customObjects = null;
+    private static final Logger LOG = LoggerFactory.getLogger(EmailProcessor.class);
+
+    @Before
+    public void setUp() throws Exception {
+        customObjects = new ArrayList<CustomObject<Email>>();
+        emailProcessor = Mockito.mock(EmailProcessor.class);
+        Mockito.doCallRealMethod().when(emailProcessor).processEmails(Mockito.any(TenantConfiguration.class));
+        tenantConfiguration = new TenantConfiguration();
+        tenantConfiguration.setProjectKey("testproject");
+        //tenantConfiguration.setClient(mockSphereClient());
+    }
+
+    @Test
+    public void shouldProcessEmails() throws Exception {
+        customObjects.add(createCustomObject("1", "pending", Statistics.RESPONSE_CODE_SUCCESS));
+        customObjects.add(createCustomObject("2", "pending", Statistics.RESPONSE_ERROR_TEMP));
+        customObjects.add(createCustomObject("3", "error", Statistics.RESPONSE_ERROR_TEMP));
+        customObjects.add(createCustomObject("4", "pending", Statistics.RESPONSE_ERROR_PERMANENT));
+        tenantConfiguration.setClient(mockSphereClient(customObjects));
+        Statistics statistic = emailProcessor.processEmails(tenantConfiguration);
+        assertEquals(statistic.getProcessedEmail(), 3);
+        assertEquals(statistic.getSuccessful(), 1);
+        assertEquals(statistic.getTempError(), 1);
+        assertEquals(statistic.getPermanentError(), 1);
+    }
+
+    @Test
+    public void shouldProcessEmails2() throws Exception {
+        customObjects.add(createCustomObject("1", "pending", Statistics.RESPONSE_CODE_SUCCESS));
+        customObjects.add(createCustomObject("2", "pending", Statistics.RESPONSE_ERROR_TEMP));
+        customObjects.add(createCustomObject("3", "pending", Statistics.RESPONSE_ERROR_TEMP));
+        customObjects.add(createCustomObject("4", "pending", Statistics.RESPONSE_ERROR_PERMANENT));
+        tenantConfiguration.setClient(mockSphereClient(customObjects));
+        Statistics statistic = emailProcessor.processEmails(tenantConfiguration);
+        assertEquals(statistic.getProcessedEmail(), 4);
+        assertEquals(statistic.getSuccessful(), 1);
+        assertEquals(statistic.getTempError(), 2);
+        assertEquals(statistic.getPermanentError(), 1);
+    }
+
+
+    @Test
+    public void shouldProcessEmails_all_successfull() throws Exception {
```

Comment:
I think this test is also not necessary and you can test all the 3 tests in one test.

Reply:
Same question as above.

Reply:
b7b2ba68262bbdd2f0be45f2f06bad1b1861b4fe